great_expectations/core/validation_config.py

```diff
@@ -215,14 +216,19 @@ def run(
             validation_result_id,
         ) = self._get_expectation_suite_and_validation_result_ids(validator)
 
-        store_validation_results(
+        ref = store_validation_results(
             validation_result_store=self._validation_results_store,
             suite_validation_result=results,
             suite_validation_result_identifier=validation_result_id,
             expectation_suite_identifier=expectation_suite_identifier,
             cloud_mode=self._validation_results_store.cloud_mode,
         )
 
+        # TODO(cdkini): Don't love the placement but this gets the URL from Cloud
+        if isinstance(ref, GXCloudResourceRef):
+            result_url = ref.response["data"]["attributes"]["validation_result"]["display_url"]
+            results.result_url = result_url
```

Comment:
This feels quite bad - where does this logic go? Probably something Cloud-specific but our encapsulation leaves a lot to be desired.

I think cleaning up the actions and other code from v0.18 will allow an opportunity to place this in a more relevant file but I'd like to discuss what makes the most sense

Reply:
Yeah I agree, this isn't great for our abstractions. It might make more sense for the store to own this logic, and the ValidationDefinition can retrieve it via `Store.get_result_url(ValidationResult)` or something. It's still not perfect, but at least cloud specific abstractions aren't leaking here.